src/dfx/src/commands/extension/list.rs

```diff
@@ -0,0 +1,41 @@
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+
+use clap::Parser;
+
+use std::io::Write;
+
+#[derive(Parser, Debug)]
+pub struct ListOpts {
+    #[clap(use_value_delimiter = true, value_delimiter = ',')]
+    /// Comma-separated list of fields from extension's metadata.json to display.
+    metadata: Vec<String>,
```

Comment:
I would just leave this metadata stuff out, if it's going to otherwise go in unfinished.  Maybe it makes more sense in a different command, displaying information for a single extension, rather than the `extension list` command.